libs/blocks/merch/merch.js

```diff
@@ -1,243 +1,138 @@
-  a.setAttribute('is', 'checkout-link');
-  const classes = ['con-button'];
-  if (document.querySelector('.marquee')) {
-    if (link.closest('.marquee')) {
-      classes.push('button-l');
-    }
-  }
-  if (link.firstElementChild?.tagName === 'STRONG' || link.parentElement?.tagName === 'STRONG') {
-    classes.push('blue');
-  }
-  a.setAttribute('class', classes.join(' '));
-  Object.assign(a.dataset, dataAttrs);
-  a.textContent = link.textContent?.replace(CTA_PREFIX, '');
-  window.tacocat.imsCountryPromise.then((countryCode) => {
-    if (countryCode) {
-      a.dataset.imsCountry = countryCode;
-    }
-  })
-    .catch(() => { /* do nothing */ });
-  return a;
-}
-
-function buildPrice(dataAttrs = {}) {
-  const span = document.createElement('span', { is: 'inline-price' });
-  span.setAttribute('is', 'inline-price');
-  Object.assign(span.dataset, omitNullValues(dataAttrs));
-  return span;
+export async function initService() {
+  const commerce = await import('../../deps/commerce.js');
+  return commerce.init(() => ({
+    ...getConfig(),
+    commerce: { priceLiteralsURL },
+  }));
 }
 
-function isCTA(type) {
-  return type === 'checkoutUrl';
+export async function getCommerceContext(el, params) {
+  const wcsOsi = params.get('osi');
+  if (!wcsOsi) return null;
+  const perpetual = params.get('perp') === 'true' || undefined;
+  const promotionCode = (
+    params.get('promo') ?? el.closest('[data-promotion-code]')?.dataset.promotionCode
+  ) || undefined;
+  return { promotionCode, perpetual, wcsOsi };
 }
 
 /**
  * Checkout parameter can be set Merch link, code config (scripts.js) or be a default from tacocat.
  * To get the default, 'undefinded' should be passed, empty string will trigger an error!
  *
  * clientId - code config -> default (adobe_com)
- * checkoutType - merch link -> metadata -> default (UCv3)
+ * workflow - merch link -> metadata -> default (UCv3)
  * workflowStep - merch link -> default (email)
  * marketSegment - merch link -> default (COM)
- * @param {*} searchParams link level overrides for checkout parameters
+ * @param {{params: URLSearchParams, service: {settings: object}}} context
  * @returns checkout context object required to build a checkout url
  */
-export function getCheckoutContext(searchParams, config) {
-  const { commerce } = config;
-  const checkoutClientId = commerce?.checkoutClientId;
-  const checkoutWorkflow = searchParams.get('checkoutType') ?? getMetadata('checkout-type');
-  const checkoutWorkflowStep = searchParams
+export async function getCheckoutContext(el, params) {
+  const context = await getCommerceContext(el, params);
+  if (!context) return null;
+  const { settings } = await initService();
+  const { checkoutClientId } = settings;
+  const checkoutMarketSegment = params.get('marketSegment');
+  const checkoutWorkflow = params.get('workflow')
+    ?? getMetadata('checkout-workflow')
+    ?? settings.checkoutWorkflow;
```

Comment:
can be changed to settings?.checkoutWorkflow

Reply:
No, per commerce library design, `settings. checkoutWorkflow` will be always set to a configured or default value (UCv3).